### DISCUSSION TITLE
How could single content files and content folders with index be sorted correctly by weight?

### DISCUSSION THREAD

**Discussion by abovewave** [adityatelange/hugo-PaperMod#950] [Q&A]
here is a example diagram of my hugo site:
.
├── content/
│   └── posts/
｜      ├── a/
｜	  ｜ ├──_index.md
｜	  ｜ ├──1.md
｜      ｜ └──2.md
｜      ├── b/
｜	  ｜ ├──_index.md
｜	  ｜ ├──3.md
｜      ｜ └──4.md
｜      ├── 5.md
｜	  └──	6.md

I set weight at the front matter of every markdown file in this order: 

a/_index > 1 > 2 > b/index > 3 > 4 > 5 >6

For the page www.example.com/posts, I am expacting that the listing would be in order

- a
- b
- 5
- 6

but what the site give me is:

- 5
- 6
- a
- b

Is there any way that the listing would be in the order I want? How could we weight across single content page and content folder page?

**Comment by heydc7** [comment]
Hey, is it resolved?

**Comment by gregsaun** [comment]
Hi all,

First of all, thanks a lot for the hard work on PaperMod!

I'm having the same issue with only folder but a mix of singles/lists. I have the following content with the weight attribute in frontmatter set as following
```
content/
    mypages/
        single-one/index.md       <--- weight: 20
        single-two/index.md       <--- weight: 40
        list-one/_index.md        <--- weight: 10
        list-two/_index.md        <--- weight: 30
        _index.md
```

When I visit mypages it shows first all the single pages (correctly ordered) and then all the list pages (correctly ordered too). This is not corresponding to the order set by the weight attribute.

For testing, I've overwritten _default/list.xml and replaced the line 55:
```{{- range $index, $page := $paginator.Pages }}```
by:
```{{- range $index, $page := $paginator.Pages.ByWeight }}```
and now the pages are correctly ordered.

Is this a bug I should report or I'm missing something? How could I solve this without overwriting list.xml?

Thanks for the help :)
